apps/rule-manager/app/controllers/RulesController.scala

```diff
@@ -121,4 +121,78 @@ class RulesController(
           )
     }
   }
+
+  def batchUpdate() = ApiAuthAction { implicit request =>
+    hasPermission(request.user, PermissionDefinition("manage_rules", "typerighter")) match {
+      case false => Unauthorized("You don't have permission to edit rules")
+      case true =>
+        val jsonBody = request.body.asJson
+        jsonBody match {
+          case Some(json) =>
+            val validatedRequest = json.validate[BatchUpdateRequest]
+            validatedRequest.fold(
+              errors => {
+                val errorMessages = errors.flatMap { case (path, validationErrors) =>
+                  validationErrors.map(error => s"${path.toString}: ${error.message}")
+                }
+                BadRequest(
+                  Json.obj(
+                    "errors"
+                      -> errorMessages
+                  )
+                )
+              },
+              batchUpdateRequest => {
+                val ids = batchUpdateRequest.ids
+                if (ids.isEmpty) BadRequest("No rule IDs provided")
+                else {
+                  val updatedRuleForm = UpdateRuleForm(
+                    category = batchUpdateRequest.fields.category,
+                    tags = batchUpdateRequest.fields.tags
+                  )
+                  val filledForm = UpdateRuleForm.form.fill(updatedRuleForm)
```

Comment:
If we pass categories and tags directly to the batch update method, then don't we have to pass category and tags in the query params i.e. `/routes/batch/:category/:tags`? My concern is then tags could end up being very long, that's why I thought it's better to put them in the request body. But this could be my incomplete Scala/Play understanding!

Reply:
I think this comment above was directed at the arguments in the function (I cannot now see the history) – you're absolutely right, we wouldn't want to specify them in the route for a few reasons, one being URL length.

The updated code with the form looks great.